sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/TaskGroup.java

```diff
@@ -399,7 +400,15 @@ public void addPostRunDependentTaskGroup(TaskGroup dependentTaskGroup) {
             }
             readyTaskEntry = super.getNext();
         }
-        return Flux.mergeDelayError(32, observables.toArray(new Flux[0]));
+        return Flux.mergeOrdered(32,
+            (some, other) -> {
+                if (rootTaskEntry.key().endsWith(some.key())) { // rootTaskEntry will be proxy-<UUID>
+                    return 1;
+                } else {
+                    return -1;
```

Comment:
Do we need a better compare? Not sure if we need to follow a<b, b<c => a<c

Reply:
I think no, the comparator is just used to select the minimal one.
![image](https://user-images.githubusercontent.com/15213781/92877984-43bf9480-f43e-11ea-8555-2e7f81298656.png)

Reply:
Why not still do a standard String compare? At least that will be deterministic based on key, instead of random based on which is left, which is right.

Reply:
Updated, but I think using String compare will still cause some `a<b, b<c but a > c` or `a<b but b<a`.

Reply:
String compare should be lexical order.

Reply:
but the first two compare is not could be swap (when these two keys are boot rootTaskEntry)

Reply:
Then you can return 0 as equal is both are root and key are same. If both are root I don't think you need to decide, any one will do.